### PR TITLE
Log bar autorange fix

### LINKED ISSUE #1352: Autorange for log plot doesn't show all the data
When using `autorange: true` with `type: "log"`, the default scaling does not show all of the data. The example is at https://jsfiddle.net/crq1tkjy

screen shot 2017-02-03 at 14 28 42

I expected to be able to see both the maximum and minimum values by default, instead anything over 10k is off the top of the graph.

### PR DESCRIPTION
**Pull request by alexcjohnson**
fixes #1352 - makes autorange work with bar traces on log axes

@etpinard please review

### CODE REVIEW

File: src/traces/bar/set_positions.js [hunk 1]
```diff
@@ -574,32 +579,13 @@ function normalizeBars(gd, sa, sieve) {
...
-                    padded = true;
-                    sMin = barBase;
-                }
-                if(barBase > sMax + sTiny) {
-                    padded = true;
-                    sMax = barBase;
-                }
-            }
+            maybeExpand(barTop);
+            maybeExpand(barBase);
```
Comment (etpinard) [inline src/traces/bar/set_positions.js]: very nice simplification.

File: test/jasmine/tests/bar_test.js [hunk 2]
```diff
@@ -697,6 +697,47 @@ describe('Bar.setPositions', function() {
...
+            {y: [1, 10, 1e10, -1]},
+            {y: [2, 20, 2e10, -2]}
+        ], {
+            yaxis: {type: 'log'},
+            barmode: 'stack',
+            barnorm: 'percent'
+        });
+
+        var ya = gd._fullLayout.yaxis;
+        expect(Axes.getAutoRange(ya)).toBeCloseToArray([1.496, 2.027], undefined, '(ya.range)');
```
Comment (etpinard) [inline test/jasmine/tests/bar_test.js]: Nice tests! I wouldn't be opposed to adding an bar log image test too if you're up for it.

**Comment by etpinard** [comment]
:dancer: image test or not. 

Thanks a lot!

File: test/jasmine/tests/bar_test.js (hunk 2 above)
Comment (alexcjohnson) [inline test/jasmine/tests/bar_test.js]: Could be nice, but for now I think I'd better move on.

**Pull request merged**